downstream.go

```diff
@@ -1517,6 +1537,12 @@ func (dc *downstreamConn) handleMessageRegistered(msg *irc.Message) error {
 				Params:  []string{upstreamName, text},
 			}
 			uc.produce(upstreamName, echoMsg, dc)
+
+			if uch, ok := uc.channels[upstreamName]; ok {
```

Comment:
Maybe we could have a `upstreamConn.updateChannelAutoDetach(name)` helper that does just that?